vm/runtime/src/lib.rs

```diff
@@ -237,6 +237,46 @@ pub trait Syscalls {
 
         Ok(())
     }
+
+    /// Verifies a proof of spacetime.
+    fn verify_winning_post(
+        &self,
+        verify_info: &WinningPoStVerifyInfo,
+    ) -> Result<(), Box<dyn StdError>> {
+        type ReplicaMapResult = Result<(SectorId, PublicReplicaInfo), String>;
+
+        // collect proof bytes
+        let proofs = &verify_info.proofs.iter().fold(Vec::new(), |mut proof, p| {
+            proof.extend_from_slice(&p.proof_bytes);
+            proof
+        });
+
+        // collect replicas
+        let replicas = verify_info
+            .challenge_sectors
+            .iter()
+            .map::<ReplicaMapResult, _>(|sector_info: &SectorInfo| {
+                let commr = cid_to_replica_commitment_v1(&sector_info.sealed_cid)?;
+                let replica = PublicReplicaInfo::new(
+                    sector_info.proof.registered_window_post_proof()?.into(),
+                    commr,
+                );
+                Ok((SectorId::from(sector_info.sector_number), replica))
+            })
+            .collect::<Result<BTreeMap<SectorId, PublicReplicaInfo>, _>>()?;
+
+        // construct prover id
+        let mut prover_id = ProverId::default();
+        let prover_bytes = verify_info.prover.to_be_bytes();
+        prover_id[..prover_bytes.len()].copy_from_slice(&prover_bytes);
+
+        // verify
+        if !verify_winning_post(&verify_info.randomness.0, &proofs, &replicas, prover_id)? {
+            return Err("Proof was invalid".to_string().into());
+        }
+
+        Ok(())
+    }
```

Comment:
This isn't a syscall, why is this being added here? What is this referencing?

Reply:
In lotus this methods seems to live in the sync struct

https://github.com/filecoin-project/lotus/blob/master/chain/sync.go#L754

Reply:
yeah, and you added that one in the PR, but this block here seems unnecessary?

Reply:
Oh I saw it in the otherlinks on the issues part and I assumed both needed to be implemented.

 I can remove it

https://github.com/ChainSafe/forest/issues/471

Reply:
Oh wait no this is a mistake on my part, was in the first stages of implementation and forgot to remove this.

Reply:
yeah, you added those two, but then this third function is unnecessary (there shouldn't be changes to the syscalls/runtime for these changes)